### PR TITLE
feat(ict,#5681): co-localisation inter-lentilles SAE↔J-Lens (lens_agreement.py + 9 tests) — verdict négatif honnête

### PR DESCRIPTION
**Pull request by jsboige**
## Résumé

Jambe **lens-agreement** de la triade d'événements du capstone axe-dérivée-temporelle (#7259) — la moitié **GPU-free** de l'approfondissement SAE↔J-Lens (#5681, B-A1).

Nouveau module numpy-only [`ict/lens_agreement.py`](MyIA.AI.Notebooks/IIT/ICT-Series/ict/lens_agreement.py) : **les lentilles SAE et J-Lens s'allument-elles aux MÊMES positions de token ?** Vivant dans des espaces disjoints (features SAE creuses exactes vs directions singulières J-space rang-k, cf tête-à-tête sec. 8.3), les deux lentilles n'ont pas de features comparables — la **concentration scalaire par position (Gini) est la monnaie commune**, et la co-localisation *temporelle* des ignitions est la bonne question.

- `lens_ignition_centers` — centres d'ignition (+ longueur `T`) par prompt pour une lentille (concentration → seuil quantile → `ignition_events`).
- `colocalize_lenses` — agrège les verdicts prompt par prompt sur les prompts partagés ; chaque lentille sélectionne **ses propres** features différentielles.

## Réutilisation (anti-duplication)

Le module **n'ajoute aucune statistique** : il réutilise la primitive déjà mergée `ict.workspace.event_colocalization` (**#7274** : distance plus-proche-voisin symétrisée + null par rotation circulaire, supérieure à Jaccard car tolérante aux quasi-coïncidences). #7274 câblait cette primitive pour la co-localisation **INTRA-lentille** beauty↔ignition ; ce module la câble pour la co-localisation **INTER-lentille** SAE↔J-Lens — complémentaire, pas redondante. La glue d'agrégation multi-prompts que la primitive laisse explicitement à l'appelant vit ici.

## Verdict scientifique — NÉGATIF honnête (pas de co-localisation)

Mesuré sur les fixtures 9B-Base couche 16 (2699 tokens, 20 prompts × 5 registres). Le verdict robuste **là où le test a de la puissance** est : **pas de co-localisation**.

| Config | prompts informatifs | z_mean | verdict agrégé |
|--------|--------------------:|-------:|----------------|
| q=0.9, pers=2  | **20 / 20** (0 undefined) | **−0.116** | dissociated |
| q=0.85, pers=2 | **20 / 20** (0 undefined) | **−0.145** | dissociated |
| q=0.9, pers=3 (défaut)  | 4 / 20 (16 undefined) | −0.007 | chance |
| q=0.85, pers=3 | 8 / 20 (12 undefined) | +0.206 | colocalized* |

`z > 0` = ignitions plus proches qu'au hasard (co-localisées) ; `z < 0` = plus éloignées.

**Lecture honnête :**
- **Aux seuils bien-dotés en puissance** (`pers=2`, les 20 prompts contribuent, 0 `undefined`), `z_mean` est **négatif** (−0.116 / −0.145) : les ignitions des deux lentilles sont, si quelque chose, *plus éloignées* qu'au hasard. **Jamais** co-localisées.
- \*La seule lecture `colocalized` (q=0.85 pers=3, z=+0.206) est **sous-dotée en puissance** : 12/20 prompts `undefined` (une lentille n'allume aucune ignition persistante). Elle est **reproduite par un positif fallacieux dans le CONTROL shuffled** (z=+0.150 à q=0.9 pers=3) → c'est un artefact de faible puissance, **pas** un signal.
- L'**instabilité du verdict** (chance ↔ dissociated ↔ colocalized selon le seuil) est *en soi* un constat négatif : une vraie co-localisation serait stable sur des seuils raisonnables.
- **Caveat de puissance** : J-Lens est **nettement clairsemée** en ignitions soutenues (beaucoup de prompts à `n_ign_b` ∈ {0, 1}) ; à `pers=3` le test manque de prompts informatifs. Le module reporte `n_ign_a`/`n_ign_b` par prompt et distingue les prompts `undefined` (flux vide) précisément pour rendre cette limite lisible.

**Cohérence avec le corpus ICT** : ce négatif prolonge la dissociation `emergence_gain`↔`ignition` déjà observée à ICT-24 (~17 % crédités) et le constat « espaces disjoints » du tête-à-tête ; il **reconnecte le négatif au Gate 5 de la synthèse** (la jambe temporelle discrimine orthogonalement — Φ/F covarient, la dérivée temporelle non). **Un résultat négatif est un résultat scientifique valide** ; il n'est pas maquillé en co-localisation.

## Portée & ce qui reste GPU-gaté

- **3ᵉ lentille raw-logit** (entropie vocab-level, l'autre moitié de B-A1) : **GPU-gatée**. Les fixtures committées sont des features **internes couche 16**, pas les logits de sortie ; extraire la lentille raw-logit demande un forward pass sur le 9B → **RECOVERABLE-MACHINE** (Track-P GPU, lane po-2024/ai-01), non inclus ici.
- **Capstone triade beauty↔ignition↔lens-agreement** (co-localisation à 3 voies + unification micro/macro du pli) = **déféré au livrable C** (#7259).

## Vérification

- `pytest tests/` → **452 passed, 2 skipped** (aucune régression ; les 2 skips = fixtures GPU absentes, gardés par `pytest.skip`).
- **9 nouveaux tests** [`tests/test_lens_agreement.py`](MyIA.AI.Notebooks/IIT/ICT-Series/tests/test_lens_agreement.py) : détection de rafales plantées, verdicts `colocalized`/`dissociated`/`undefined`, saut sur `T` incompatible, reproductibilité par graine, module numpy-only (aucune instruction `import torch`), intégration sur traces réelles (si présentes → 20 prompts).
- Construction synthétique déterministe : concentration baseline **alternée** (aucune fenêtre `persistence=3` ne franchit le quantile 0.9) + rafales triangulaires (pic au centre), Gini ciblé exactement via `f=(5L+1)/6` (invariant d'échelle → la mise à l'échelle par jeu pilote `differential_features` sans toucher la concentration).

Module numpy-only : aucune dépendance torch (le GPU reste confiné aux scripts d'extraction).

See #5681, #7259, #4588

### CODE REVIEW

**Comment by jsboige** [comment]
Independent GLM peer-verify (po-2023, **EXEC_PROVED** — strongest level). PR had 0 reviews.

**Ran the full suite firsthand** in a fresh worktree (`numpy 2.4.3`, `pytest 8.3.5`, GPU-free):
```
9 passed in 1.91s
```
All 9 tests PASS, including `test_real_fixtures_colocalization_runs` (NOT skipped — the committed traces `ict21_sae_layer16_trained.npz` + `ict24_jlens_layer16_trained.npz` are present, so the integration test ran on **real data**, not just synthetic).

**Reproduced the honest-negative verdict on real fixtures** (independent repro vs PR body table):

| Config | PR body claim | My repro |
|---|---|---|
| q=0.9, persistence=3 (default) | 4/20 informative, z=−0.007, chance | 4/20 (16 undefined), z=−0.0245, **chance** ✅ |
| q=0.9, persistence=2 (high power) | 20/20 informative, z=−0.116, dissociated | 20/20 (0 undefined), z=−0.1138, **dissociated** ✅ |

The high-power config matches to 2 decimals (−0.1138 vs −0.116). The default-config z (−0.0245 vs −0.007) differs within expected rotation-null sampling noise (n_null=200, only 4 informative prompts → high variance). **Verdict honesty confirmed**: no co-localization where the test has power.

**Code review (math + architecture):**
- `lens_ignition_centers`: concentration series (Gini) → per-series quantile threshold `q` → `ignition_events` → centers. Per-series thresholding is the right call — two lenses with different dynamics each keep their ~10% most concentrated positions. Sound.
- `colocalize_lenses`: each lens selects its OWN `k` differential features (disjoint spaces), then `event_colocalization` prompt-by-prompt, then majority verdict + `z_mean` over defined z's. T-mismatch → skip; empty-ignition → undefined (excluded from z). Correct guards.
- **Anti-duplication holds**: the module adds NO new statistic — it reuses `ict.workspace.event_colocalization` (#7274, NN-distance symmetrized + rotation null) and only provides the inter-lens glue the primitive leaves to the caller. Complementary to #7274 (intra-lens beauty↔ignition), not redundant.
- **numpy-only confirmed**: `test_lens_agreement_module_is_numpy_only` checks at line level (not substring — correctly avoids docstring false-positive where the prose says « aucun import torch »).

**Scope of this check (honest):** I verified module logic + test execution + real-fixture verdict reproduction + anti-duplication. I did NOT re-derive the Dehaene/Grün theoretical framing or audit the upstream `event_colocalization` primitive (#7274, already merged & tested in `test_workspace.py`).

No CHANGES_REQUESTED — sound engineering, honest negative verdict reproduced firsthand. (Worker po-2023 does not merge/approve; informational peer-verify.)
